pinot-core/src/main/java/org/apache/pinot/core/query/scheduler/QueryScheduler.java

```diff
@@ -237,6 +240,13 @@ protected byte[] processQueryAndSerialize(ServerQueryRequest queryRequest, Execu
       _serverMetrics.addTimedTableValue(tableNameWithType, ServerTimer.TOTAL_CPU_TIME_NS, totalCpuTimeNs,
           TimeUnit.NANOSECONDS);
     }
+    if (queryHasMVSelectionOrderBy) {
+      _serverMetrics.addMeteredTableValue(tableNameWithType, ServerMeter.QUERY_HAS_MV_SELECTION_ORDER_BY, 1);
+    }
+
+    // Remove the 'QUERY_HAS_MV_SELECTION_ORDER_BY' metadata from the DataTable as the Broker does not need to know
+    // about this metadata or emit a metric related to it
+    dataTableMetadata.remove(MetadataKey.QUERY_HAS_MV_SELECTION_ORDER_BY.getName());
```

Comment:
I am wondering if we need to flag the offending query by sending the metadata to the broker and flagging the query in the broker log?. if the metric is hit, we can search the log, find the offending query, and ask the users to clean up these queries? If the mechanism is made generic enough, it could be used for other such usecases as well?

Reply:
Thanks for the suggestion! What do you mean by flagging the query on the Broker side? 

@siddharthteotia and I did discuss whether we need a Broker side metric for this before opening the PR. We can always look for the metric on the servers and check the server logs for the broker and check the queries coming in at the time on the Broker side to identify what these queries are. So we didn't think it was necessary to add a metric on the Broker side too. Let me know if you think there are other advantages of adding this on the Broker side too.

Reply:
Great suggestion @amrishlal I've updated the PR to add the log too.